gulpfile.js

```diff
@@ -55,6 +65,48 @@ gulp.task('images', function buildImages() {
     .pipe(gulp.dest(paths.images.dest));
 });
 
+/**
+ * Adds necessary AMP script tags.
+ *
+ * Replaces the placeholder '${ampjs}' with the AMP base script and any
+ * necessary scripts to support custom elements.
+ *
+ * @param {!Vinyl} file The file to scan and add tags to.
+ * @return {!Vinyl} The modified file.
+ */
+function addScriptTags(file) {
+  const dom = new JSDOM(file.contents.toString());
```

Comment:
Wondering if we need JSDOM for this, or if simply searching through the file for a string would suffice.  But I suppose your way avoids some pitfalls... like seeing a component tag in a comment...

Reply:
Yeah, I think as I am after element names, JSDOM is right for this, to avoid picking up tags in comments or other edge cases, as you mention.